resp/bench_test.go

```diff
@@ -0,0 +1,120 @@
+package resp
+
+import (
+	"bufio"
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func BenchmarkIntUnmarshalRESP(b *testing.B) {
+	tests := []struct {
+		Input string
+	}{
+		{"-1"},
+		{"-123"},
+		{"1"},
+		{"123"},
+		{"+1"},
+		{"+123"},
+	}
+
+	for _, test := range tests {
+		input := ":" + test.Input + "\r\n"
```

Comment:
In the other benches you do a `test := test` here, is that necessary here as well?

Reply:
The `test := test` makes sure that the closure passed to b.Run uses the correct `test` value since closures capture references and the referenced value changes with the next iteration. Since this test doesn't reference `test` inside the closure there is no need to make a copy of `test` in the block.